core/genesis.go

```diff
@@ -457,8 +462,15 @@ func (g *Genesis) configOrDefault(ghash common.Hash) *params.ChainConfig {
 
 // ToBlock returns the genesis block according to genesis specification.
 func (g *Genesis) ToBlock() *types.Block {
-	root, err := g.Alloc.deriveHash()
-	if err != nil {
+	var root common.Hash
+	var err error
+	if g.StateHash != nil {
+		if len(g.Alloc) > 0 {
+			panic(fmt.Errorf("cannot both have genesis hash %s "+
+				"and state-allocation with %d entries", *g.StateHash, len(g.Alloc)))
```

Comment:
I wonder if the error message should be saying you can't have any state-allocations. Minor thing but feels weird to say you can't have 5 entries, then if you remove one it will tell you that 4 isn't allowed either.

Reply:
will update this